cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,297 @@
+        jobs = sorted(jobs, key=lambda x: x.path_local)
+
+        # Final go from user & transfer
+        itransfer = iRODSTransfer(jobs, ask=not self.args.yes)
+        logger.info("Planning to transfer the following files:")
+        for job in jobs:
+            output_logger.info(job.path_local)
+        logger.info(f"With a total size of {sizeof_fmt(itransfer.size)}")
+        logger.info("Into this iRODS collection:")
+        output_logger.info(f"{self.target_coll}/")
+
+        if not self.args.yes:
+            if not input("Is this OK? [y/N] ").lower().startswith("y"):  # pragma: no cover
+                logger.info("Aborting at your request.")
+                sys.exit(0)
+
+        itransfer.put(recursive=self.args.recursive, sync=self.args.sync)
+        logger.info("File transfer complete.")
+
+        # Compute server-side checksums
+        if self.args.remote_checksums:  # pragma: no cover
+            logger.info("Computing server-side checksums.")
+            itransfer.chksum()
+
+    def build_file_list(self) -> typing.List[typing.Dict[Path, Path]]:
+        """
+        Build list of source files to transfer.
+        iRODS paths are relative to target collection.
+        """
+
+        source_paths = [Path(src) for src in self.args.sources]
+        output_paths = list()
+
+        for src in source_paths:
+            try:
+                abspath = src.resolve(strict=True)
+            except FileNotFoundError:
+                logger.warning(f"File not found: {src.name}")
+                continue
+            except RuntimeError:
+                logger.warning(f"Symlink loop: {src.name}")
+                continue
+
+            excludes = self.args.exclude
+            if src.is_dir():
+                paths = abspath.glob("**/*" if self.args.recursive else "*")
+                for p in paths:
+                    if excludes and any([p.match(e) for e in excludes]):
+                        continue
+                    if p.is_file() and not p.suffix.lower() == ".md5":
+                        output_paths.append({"spath": p, "ipath": p.relative_to(abspath)})
+            else:
+                if not any([src.match(e) for e in excludes if e]):
+                    output_paths.append({"spath": src, "ipath": Path(src.name)})
+        return output_paths
+
+    def build_jobs(self, source_paths: typing.Iterable[Path]) -> typing.Set[TransferJob]:
+        """Build file transfer jobs."""
+
+        transfer_jobs = []
+
+        for p in source_paths:
+            path_remote = f"{self.target_coll}/{str(p['ipath'])}"
+            md5_path = p["spath"].parent / (p["spath"].name + ".md5")
+
+            if md5_path.exists():
+                logger.info(f"Found md5 hash on disk for {p['spath']}")
+            else:
+                md5sum = compute_md5_checksum(p["spath"])
+                with md5_path.open("w", encoding="utf-8") as f:
+                    f.write(f"{md5sum}  {p['spath'].name}")
```

Comment:
md5 computation could be made controllable by user (with default = on)

Reply:
Why would the user want to turn it off?

Reply:
One might expect to have all md5 files already and not want to upload 'incomplete' file sets

Reply:
Still not getting it. What kind of incomplete file sets?